### DISCUSSION TITLE
Writing back to host without creating airflowtmp folder

### DISCUSSION THREAD

**Discussion by fafa92** [apache/airflow#31076] [General]
Hi,

I'm trying to write back results from dockeroperator back to the host machine with these two arguments "host_tmp_dir" and "tmp_dir". It seems for each task/operator airflow creates a new folder that starts with "airflowtmpRANDOMTEXT" on the host. If I want to make a DAG that each step gets the last step output as the input for the next step it appears this approach is not doable to be done on the fly since you need to know the RANDOMTEXT each time for the next task. Is it possible for airflow somehow to write back to the host path that's been referenced without creating a new folder each time and without using Xcom in this process? 

I really appreciate any help in advance.

**Comment by potiuk** [comment]
You can write your custom version of the operator if you really need it - this is normal in Airflow and first-class citizen that if you miss something in an operator, you can extend it and replace part of it that you need to adapt. 

But also I think you should be aware (and one of the reasons why it does not work this way) that the approach of sharing the output wiht another task input via file is not a "generally usable" solution. It might work for you if you use LocalExecutor and single instance of Scheduler, but this is very "limited" way Airflow can be used. In general case - Airflow is a disstributed scheduling system - any task can be executed on any machine and you have no control over it really. This also means that airflow's operators can only share the data in an easy way via XCom not via "shared files on the local filesystem". 

XCom mechanism takes care about serializing the data and storing it - either in the DB or remote storage so that it can be shared with other tasks. This is what regular "output" from the DockerOperator (and any other)  gets automatically sent to. you can also use Xcom manually in your operator to push any data there - and pull it when your task needs it.
